segmentation_labeling_app/transforms/transform_pipeline.py

```diff
@@ -0,0 +1,204 @@
+    quantile = argschema.fields.Float(
+        required=True,
+        default=0.1,
+        description=("quantile threshold for outlining an ROI. "))
+    input_fps = argschema.fields.Int(
+        required=False,
+        default=31,
+        description="frames per second of input movie")
+    output_fps = argschema.fields.Int(
+        required=False,
+        default=4,
+        description="frames per second of downsampled movie")
+    downsampling_strategy = argschema.fields.Str(
+        required=False,
+        default="average",
+        validator=mm.validate.OneOf(['average', 'first', 'last', 'random']),
+        description="what downsampling strategy to apply to movie and trace")
+    random_seed = argschema.fields.Int(
+        required=False,
+        default=0,
+        description="random seed to use if downsampling strategy is 'random'")
+
+    @mm.pre_load
+    def set_segmentation_run_id(self, data, **kwargs):
+        """if segmentation_run_id not provided, query the database
+        using the other provided args
+        """
+        if "segmentation_run_id" not in data:
+            if not all(i in data for i in [
+                    "ophys_experiment_id",
+                    "ophys_segmentation_commit_hash"]):
+                raise TransformPipelineException(
+                        "if omitting arg segmentation_run_id, must "
+                        "include args ophys_experiment_id and "
+                        "ophys_segmentation_commit_hash")
+
+            db_credentials = query_utils.get_labeling_db_credentials()
+            db_connection = query_utils.DbConnection(**db_credentials)
+            query_string = (
+                "SELECT id FROM segmentation_runs WHERE "
+                f"ophys_experiment_id={data['ophys_experiment_id']} AND "
+                "ophys_segmentation_commit_hash="
+                f"'{data['ophys_segmentation_commit_hash']}'")
+            entries = db_connection.query(query_string)
+            if len(entries) != 1:
+                raise TransformPipelineException(
+                    f"{query_string} did not return exactly 1 result")
+            data['segmentation_run_id'] = entries[0]['id']
+        return data
+
+
+class TransformPipeline(argschema.ArgSchemaParser):
+    default_schema = TransformPipelineSchema
+
+    def run(self, db_conn: query_utils.DbConnection):
+        output_dir = Path(self.args['artifact_basedir']) / \
+                     f"segmentation_run_id_{self.args['segmentation_run_id']}"
+        os.makedirs(output_dir, exist_ok=True)
+
+        # get all ROI ids from this segmentation run
+        query_string = ("SELECT id FROM rois WHERE segmentation_run_id="
+                        f"{self.args['segmentation_run_id']}")
+        entries = db_conn.query(query_string)
+        roi_ids = [i['id'] for i in entries]
+
+        # NOTE: here could be a good place to put a pre-filtering stepw
```

Comment:
See above with `# TODO` for visibility

Reply:
changed NOTEs to TODOs

Reply:
after discussion: we are deciding to not implement a pre-filter at this time. This TODO will remain.